pkg/pillar/zedcloud/deferred.go

```diff
@@ -223,46 +237,61 @@ func (ctx *DeferredContext) handleDeferred(log *base.LogObject, event time.Time,
 func SetDeferred(zedcloudCtx *ZedCloudContext, key string, buf *bytes.Buffer,
 	size int64, url string, bailOnHTTPErr, withNetTracing bool, itemType interface{}) {
 
-	zedcloudCtx.deferredCtx.setDeferred(zedcloudCtx, key, buf, size, url, bailOnHTTPErr,
-		withNetTracing, itemType)
+	item := CreateDeferred(key, buf, size, url, bailOnHTTPErr,
+		withNetTracing, itemType, DeferredItem_NoFlags)
+	zedcloudCtx.deferredCtx.addDeferred(zedcloudCtx, item)
+}
+
+// AddDeferredList - schedules list of deferred requests
+func AddDeferredList(zedcloudCtx *ZedCloudContext, items []*DeferredItem) {
+	for _, item := range items {
+		zedcloudCtx.deferredCtx.addDeferred(zedcloudCtx, item)
+	}
 }
 
-func (ctx *DeferredContext) setDeferred(zedcloudCtx *ZedCloudContext,
-	key string, buf *bytes.Buffer, size int64, url string, bailOnHTTPErr,
-	withNetTracing bool, itemType interface{}) {
+func (ctx *DeferredContext) addDeferred(zedcloudCtx *ZedCloudContext,
```

Comment:
I didn't check if both AddDeferred and SetDeferred are still used, but they intentionally are different since Set replaces an item with the same key and Add does not. So renaming "set"to "add" while keeping the replace semantics of "set" doesn't make sense to me.

Reply:
Turns out that AddDeferred (queue without replacement) is gone except one mention in a comment.
But since your new AddDeferredList has replacement semantics it makes sense naming it SetDeferredList, or carefully change all comments around the replacement semantics to make sure that we state that Add is doing a replace if the key already exists.